### PR TITLE
UML-169 - create PR status based on code coverage

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
Moves upload of test results to codecov to a single job that runs in parallel to terraform applies
Some uploads have taken longer so this aims to reduce overall workflow time.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -193,9 +198,14 @@ orbs:
...
               path: build/service-front/
-          - run:
-              name: Upload to Codecov
-              command: bash <(curl -s https://codecov.io/bash) -t $CODECOV_TOKEN
+          - persist_to_workspace:
+              root: build
+              paths:
+                - service-front/
+              when: always
+          # - run:
```
Comment (cooperaj) [inline .circleci/config.yml]: Can probably ditch this

File: .circleci/config.yml [hunk 2]
```diff
@@ -286,9 +296,14 @@ orbs:
...
               path: build/service-api/
-          - run:
-              name: Upload to Codecov
-              command: bash <(curl -s https://codecov.io/bash) -t $CODECOV_TOKEN
+          - persist_to_workspace:
+              root: build
+              paths:
+                - service-api/
+              when: always
+          # - run:
```
Comment (cooperaj) [inline .circleci/config.yml]: and this

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+coverage:
```
Comment (cooperaj) [inline codecov.yml]: We need to have a bit of a think about the setting for this

Comment (cooperaj) [reply codecov.yml]: CodeCov provides two views of your testing metrics.

Comment (cooperaj) [reply codecov.yml]: A project level view, and this should be fairly obvious for most. It defines what the coverage is for the project as a whole. We might decide we want something sensible like 80% coverage in the end, but sadly we're not their yet so will have to bump up to that number. For any given PR we want our  project coverage to stay the same, or go up.

Comment (cooperaj) [reply codecov.yml]: A patch level view. This took me a while to figure. This is the percentage of code in your PR that is covered by tests. Us, being good developers should want this as close to 100% as we can get it. i.e. all things we do are tested. Had this been in from the beginning we wouldn't have an issue meeting the 80% requirement for the project.

But this makes the refactoring of older, untested, code a pain as it means you *need* to write tests for that code as well as make whatever changes you need. It's good practise to do it anyway...

Comment (andrewpearce-digital) [reply codecov.yml]: Set this for project
target: auto
threshold: 5%

will allow a 5% drop on the current master value.

File: .circleci/config.yml (hunk 1 above)
Comment (andrewpearce-digital) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 2 above)
Comment (andrewpearce-digital) [reply .circleci/config.yml]: done

File: codecov.yml (hunk 3 above)
Comment (andrewpearce-digital) [reply codecov.yml]: I've added this for each service on the project

**Pull request merged**